src/display/canvas.js

```diff
@@ -1417,18 +1417,53 @@ var CanvasGraphics = (function CanvasGraphicsClosure() {
       return shadow(this, 'isFontSubpixelAAEnabled', enabled);
     },
 
-    showText: function CanvasGraphics_showText(glyphs) {
+    showText: function CanvasGraphics_showText(unmergedGlyphs) {
       var current = this.current;
       var font = current.font;
       if (font.isType3Font) {
-        return this.showType3Text(glyphs);
+        return this.showType3Text(unmergedGlyphs);
       }
 
       var fontSize = current.fontSize;
       if (fontSize === 0) {
         return;
       }
 
+      // If glyphs placed next to each other are of the same type,
+      // we can merge them and reduce calls to fillText/drawChar.
+      var glyphs = unmergedGlyphs.reduce((glyphList, glyph, index) => {
+        if (index < 1) {
+          glyphList.push(glyph);
+          return glyphList;
+        }
+
+        var previousGlyph = glyphList[glyphList.length - 1];
+
+        if (isNum(previousGlyph) !== isNum(glyph) ||
+            glyph.accent ||
+            glyph.isInFont !== previousGlyph.isInFont ||
+            glyph.isSpace !== previousGlyph.isSpace
+        ) {
```

Comment:
Please move this line back to the previous line as well, so we get `glyph.isSpace !== previousGlyph.isSpace) {` on one line.